### PR TITLE
Make KeywordsVocabulary Work Smoothly with Unicode Vocabularies.

### PR DESCRIPTION
**Pull request by l34marr**
Fix Issue #15.

### CODE REVIEW

File: plone/app/vocabularies/catalog.py [hunk 1]
```diff
@@ -432,16 +432,14 @@ def __call__(self, context, query=None):
...
-                term = term.encode('utf-8')
-            return term
+            return safe_unicode(term)
 
         # Vocabulary term tokens *must* be 7 bit values, titles *must* be
         # unicode
+        # no need to use portal encoding for transitional encoding from
+        # unicode to ascii. utf-8 should be fine.
         items = [
-            SimpleTerm(i, b2a_qp(safe_encode(i)), safe_unicode(i))
```
Comment (jensens) [inline plone/app/vocabularies/catalog.py]: you may want to remove the now superfluos import of b2a_qp at line 4

**Comment by jensens** [comment]
Please add a line to CHANGES.rst

**Comment by jensens** [comment]
http://jenkins.plone.org/job/pull-request-5.0/124/ tests are running here

**Comment by jensens** [comment]
tests are failing, needs fix

**Comment by jensens** [comment]
I'd like this change to be reviewed by someone else too. I'am not sure about the overall impact on already stored values. Afaik the value is stored, the token is only temporary (for the form itself). May someone confirm this? If this is true and tests are passing we can merge it.

I also started a new PR test here: http://jenkins.plone.org/job/pull-request-5.0/131/

**Comment by davisagli** [comment]
This doesn't look right; ITokenizedTerm in zope.schema says of the token that "the value of this attribute must be a non-empty 7-bit string." It's meant to be something that can be used without further encoding/decoding.

Can you explain the problem that you're trying to solve?

**Comment by l34marr** [comment]
Specifically we spot this issue on Plone 5 + plone.app.widgets 2.0.0 + plone.app.vocabularies 2.1.17, when entering keyword terms *twice*, I mean, save and re-edit them. Please see another video, this time we try produce this issue in German. Hope this helpful. http://www.youtube.com/watch?v=CQwh3gYoicM

**Comment by jensens** [comment]
Yeah! This is the first bug report I (at least) saw in Plone with a screencast. This is a great idea and shows really the broken parts. Very helpful.

**Comment by jensens** [comment]
If the vocabulary value is not respected and instead the token is stored this is a bug in the widget. The vocabulary is fine. 
Also with latest builodut.coredev 5 following exactly the steps in the video I can not select the existing value (at 0:30). May you try again with all versions/checkouts at latest?
Thanks anyway!

**Pull request closed without merging**